onnxscript/rewriter/fuse_pad_into_conv.py

```diff
@@ -0,0 +1,286 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+"""Fuses Pad nodes into preceding nodes. Supported fusion patterns:
+- Pad ∘ Conv          -> Conv
+- Pad ∘ ConvInteger   -> ConvInteger
+
+To make some rules possible, we implicitly transform `auto_pad` attribute into its explicit list.
+"""
+
+import typing
+
+import numpy as np
+import onnx_ir as ir
+
+from onnxscript.rewriter import pattern as orp
+
+
+def fill_pads_with_axes(
+    pads: typing.Sequence[int], axes: typing.Sequence[int], rank: int
+) -> typing.List[int]:
+    new_pads = []
+    for axis in range(rank):
+        if axis not in axes:
+            start_value = end_value = 0
+        else:
+            start_value = pads[axes.index(axis)]
+            end_value = pads[axes.index(axis) + len(axes)]
+        pad_len = len(new_pads) // 2
+        new_pads.insert(pad_len + axis, end_value)
+        new_pads.insert(axis, start_value)
+    return new_pads
+
+
+def read_conv_attributes(ir_conv: ir.Node) -> dict[str, typing.Sequence[int] | str]:
+    # Read attributes
+    attributes = {}
+    if (kernel_shape := ir_conv.attributes.get("kernel_shape", None)) is not None:
+        attributes["kernel_shape"] = kernel_shape.as_ints()
+    else:
+        attributes["kernel_shape"] = ir_conv.inputs[1].shape[2:]
```

Comment:
What if the shape of `ir_conv.inputs[1]` is not known? Just making sure it is checked